openmdao/core/problem.py

```diff
@@ -2188,6 +2188,52 @@ def get_reports_dir(self, force=False):
 
         return reports_dirpath
 
+    def get_indep_vars(self, include_design_vars=True):
+        """
+        Retrieve the independent variables in the Problem.
+
+        Returns a dictionary mapping the promoted names of indep_vars which the user is
+        expected to provide to the metadata for the associated independent variable.
+
+        A output is designated as an independent variable if it is tagged with
+        'openmdao:indep_var'. This includes IndepVarComp by default, and users are
+        able to apply this tag to their own component outputs if they wish
+        to provide components with IndepVarComp-like capaability.
+
+        Parameters
+        ----------
+        include_design_vars : bool
+            If True, include design variables in the list of problem inputs.
+            The user may provide values for these but ultimately they will
+            be overwritten by the Driver.
+
+        Returns
+        -------
+        dict
+            A dictionary mapping the promoted names of all independent variables
+            in the model to their metadata.
+        """
+        model = self.model
+        if model._outputs is None:
+            raise RuntimeError("get_indep_vars requires that final_setup has been "
+                               "run for the Problem.")
+
+        connections = model._conn_global_abs_in2out
+        desvar_prom_names = set(model.get_design_vars(recurse=True, use_prom_ivc=True).keys())
```

Comment:
You don't really need to create a set here.  The dict returned from model.get_design_vars should give you what you need to do  quick checks to see if a var is a DV.